typing/src/main/java/org/navgurukul/typingguru/utils/TypingGuruPreferenceManager.kt

```diff
@@ -1,31 +1,19 @@
 package org.navgurukul.typingguru.utils
 
-import android.content.Context
 import android.content.SharedPreferences
 
-object TypingGuruPreferenceManager {
-    private const val PREF_NAME = "typing-guru"
-    private var mSharedPreferences: SharedPreferences? = null
-
-    fun init(context: Context) {
-        if (mSharedPreferences == null) {
-            mSharedPreferences = context.getSharedPreferences(
-                PREF_NAME,
-                Context.MODE_PRIVATE
-            )
-        }
-    }
+class TypingGuruPreferenceManager(private val sharedPreferences: SharedPreferences) {
 
     private val editor: SharedPreferences.Editor
-        private get() = mSharedPreferences!!.edit()
+        private get() = sharedPreferences!!.edit()
 
     fun setWebViewDisplayStatus(isShown: Boolean) {
         val editor = editor
         editor.putBoolean("isShown", isShown)
-        editor.commit()
+        editor.apply()
```

Comment:
better to use `sharedpreferences.edit { }` extension here which encapsulated boiler plate code

Reply:
changed to sharedPreferences.edit().putBoolean("isShown", isShown).apply()

Reply:
That works too, but you can replace this code with `sharedPreferences.edit(){ putBoolean("isShown", isShown) }` too. You don't need to use `apply()` in this case as it is encapsulated. Feel free to ignore it if you don't prefer it

Reply:
actually I referred PlaygroundRepositoryImpl and found same there
    override fun cacheCode(code: String) {
        sharedPreferences.edit().putString(KEY_PREF_CODE_BACKUP, code).apply()
    }.

I will try to use extension function in future PR.